tools/amd_build/pyHIPIFY/hipify-python.py

```diff
@@ -487,6 +504,34 @@ def replace_math_functions(input_string):
     return output_string
 
 
+def hip_header_magic(input_string):
+    """If the file makes kernel builtin calls and does not include the cuda_runtime.h header,
+    then automatically add an #include to match the "magic" includes provided by NVCC.
+    TODO:
+        Update logic to ignore cases where the cuda_runtime.h is included by another file.
```

Comment:
Is this really necessary? It should be harmless to include a header multiple times?

Reply:
it is harmless, I think this is strictly in the "beautification" realm.